src/app/private/admin/association-management/association-management-update/association-management-update.component.ts

```
@@ -119,41 +125,120 @@ export class AssociationManagementUpdateComponent implements OnInit {
   }
 
-  uploadRefugeImage(imageInput: any) {
-    const file: File = imageInput.files[0];
-    const reader = new FileReader();
+  selectFile(event: any): void {
+    this.selectedFile = new FileUpload((event.target.files).item(0));
+    let reader = new FileReader();
+    reader.onload = (event: any) => {
+      this.selectedFile.url = event.target.result;
+    };
+
+    reader.onerror = (event: any) => {
+      this.selectedFile.error = event.target.error.code;
+    };
+
+    reader.readAsDataURL(event.target.files[0]);
+  }
+
+  selectFileList(event: any): void {
+    this.selectedFiles.push(new FileUpload((event.target.files).item(0)));
+    let reader = new FileReader();
+    reader.onload = (event: any) => {
+      this.selectedFiles[this.selectedFiles.length - 1].url = event.target.result;
+    };
+
+    reader.onerror = (event: any) => {
+      this.selectedFiles[this.selectedFiles.length - 1].error = event.target.error.code;
+    };
+
+    reader.readAsDataURL(event.target.files[0]);
+  }
 
-    reader.addEventListener('load', (event: any) => {
+  uploadDataAndImages(refuge: Refuge): void {
+    if (refuge.id != null) {
+      this.sendImagesForRemove(refuge.logoPath, refuge.images);
+    }
+    if (this.selectedFile) {
+      this.firebaseStorageService.pushFileToStorage(this.selectedFile).subscribe(
+        percentageChanges => {
+          if (percentageChanges == 100) {
+            this.firebaseStorageService.getFileFromStorage(this.selectedFile).subscribe(downloadURL => {
+              if (downloadURL) {
+                this.selectedFile.downloadURL = downloadURL;
+                refuge.logoPath = { name: this.selectedFile.file.name, downloadURL: this.selectedFile.downloadURL };
+                refuge.images = [];
+                if (this.selectedFiles.length > 0) {
+                  this.sendImagesToStorage(refuge);
+                } else {
+                  this.sendOperation(refuge);
+                }
+              }
```

In case download url is null, what will happen?, maybe return an error state